commonv5/ts/sonarqube/Analysis.ts

```diff
@@ -44,145 +38,124 @@ const separatorStyle = `
 
-    return `<ul style="${textStyle()}">
-        ${rows.join(" \r\n").trim()}
-    </ul>\r\n\r\n`; // 2 carriage returns to prevent any malformed summary results
+    return `
+    <ul style="${textStyle()}">
+      ${rows.join(" \r\n").trim()}
+    </ul>\r\n\r\n
+    `; // 2 carriage returns to prevent any malformed summary results
   }
 
-  private getDashboardLink() {
-    if (!this.dashboardUrl) {
+  private getHtmlDashboardLink() {
+    if (!this.result.dashboardUrl) {
       return "";
     }
-    const linkText = `Detailed ${this.endpointType} report`;
-
-    return `<div style="margin-top: 16px;">
-        <a href="${this.dashboardUrl}" style="${textStyle("#5D6CD0")}">
-        ${linkText}
-        <span class="bowtie-icon bowtie-navigate-external" style="color: #5D6CD0; font-size: 20px; vertical-align: middle;"></span>
-        </a>
-      </div>`;
+
+    return `
+    <div style="margin-top: 16px;">
+      <a href="${this.result.dashboardUrl}" style="${textStyle("#5D6CD0")}">
+      Detailed ${this.endpointType} report
+      <span class="bowtie-icon bowtie-navigate-external" style="color: #5D6CD0; font-size: 20px; vertical-align: middle;"></span>
+      </a>
+    </div>
+    `;
   }
 
-  public getWarnings() {
-    if (this.warnings.some((w) => w.includes("Please update to at least Java 11"))) {
-      return `<br><span>&#9888;</span><b>${this.warnings.find((w) =>
-        w.includes("Please update to at least Java 11"),
-      )}</b>`;
+  public getHtmlWarnings() {
+    if (this.result.warnings.some((w) => w.includes("Please update to at least Java 11"))) {
+      return `
+      <br>
+      <span>&#9888;</span>
+      <b>
+        ${this.result.warnings.find((w) => w.includes("Please update to at least Java 11"))}
+      </b>
+      `;
     } else {
       return "";
     }
   }
-
-  private getQualityGateColor() {
-    const colors = {
-      OK: "#6EB712",
-      WARN: "#F5B840",
-      ERROR: "#D92D20",
-    };
-
-    return colors[this.status] ?? "#b4b4b4";
-  }
-
-  public static getAnalysis({
-    analysisId,
-    projectName,
-    endpoint,
-    metrics,
-    dashboardUrl,
-    warnings,
-  }: {
-    analysisId: string;
-    dashboardUrl?: string;
-    endpoint: Endpoint;
-    projectName?: string;
-    metrics?: Metrics;
-    warnings: string[];
-  }): Promise<Analysis> {
-    tl.debug(`[SQ] Retrieve Analysis id '${analysisId}.'`);
-    return get(endpoint, "/api/qualitygates/project_status", true, { analysisId }).then(
-      ({ projectStatus }: { projectStatus: IAnalysis }) =>
-        new Analysis(projectStatus, endpoint.type, warnings, dashboardUrl, metrics, projectName),
-      (err) => {
-        if (err && err.message) {
-          tl.error(`[SQ] Error retrieving analysis: ${err.message}`);
-        } else if (err) {
-          tl.error(`[SQ] Error retrieving analysis: ${JSON.stringify(err)}`);
-        }
-        throw new Error(`[SQ] Could not fetch analysis for ID '${analysisId}'`);
-      },
-    );
-  }
```

Comment:
This function was a bit weird to me because it had mixed semantic:
1. It fetched project status
2. It instantiates an `Analysis` object with the project status

I split it in 2:
- a `fetchProjectStatus` method with most of this code
- this method `getAnalysis(..)` that only instantiates an `Analysis` object with all the data